### DISCUSSION TITLE
Problem finding maximum element index -- How do I do this?

### DISCUSSION THREAD

**Discussion by infinite-pursuits** [Consensys/gnark#927] [Q&A]
I have a circuit with the following piece of code:

```
cmpresult := api.Cmp(logits[0], logits[1])
ind := api.Select(cmpresult, frontend.Variable(0), frontend.Variable(1))
```

This gives me the error ```constraint #3812 is not satisfied: -1 ⋅ 2 != 0``` , possibly because I use api.Cmp (can be 0,1, or -1) inside of api.Select which expects 0 or 1 only.

The functionality I want is just to find the index of the maximum element in a 2 element list. Is there an alternate way to implement this without encountering the error?

**Comment by ivokub** [comment]
You could provide maximum value using a hint, compare this value against all values inside the slice to be less equal than the maximum value (`cmpresult = Cmp(values[i], max); AssertIsEqual(cmpresult*(1-cmpresult), 0)`). Additionally, you have to show that at least one of the elements is equal to the maximum value, but you can do this by counting the number of `cmpresult == 0` and it must be bigger than 0.
